src/gen-ts.ts

```diff
@@ -94,33 +94,37 @@ function handleDocument(doc: analyzer.Document, root: ts.Document) {
   // TODO Should we traverse and serialize all features in the same order they
   // were originally declared, instead of grouping by type as we do here?
   for (const element of doc.getFeatures({kind: 'element'})) {
-    handleElementOrBehavior(element, root);
+    handleElement(element, root);
   }
   for (const behavior of doc.getFeatures({kind: 'behavior'})) {
-    handleElementOrBehavior(behavior, root);
+    handleBehavior(behavior, root);
+  }
+  for (const mixin of doc.getFeatures({kind: 'element-mixin'})) {
+    handleMixin(mixin, root);
   }
   for (const fn of doc.getFeatures({kind: 'function'})) {
     handleFunction(fn, root);
   }
 }
 
 /**
- * Add the given Element or Behavior to the given TypeScript declarations
- * document.
+ * Add the given Element to the given TypeScript declarations document.
  */
-function handleElementOrBehavior(
-    feature: analyzer.Element|analyzer.PolymerBehavior, root: ts.Document) {
-  let fullName;   // Fully qualified reference, e.g. `Polymer.DomModule`.
-  let className;  // Just the class name e.g. `DomModule`.
-  let parent;     // Where in the namespace tree does this live.
+function handleElement(feature: analyzer.Element, root: ts.Document) {
+  let isClassBased;  // Whether this is a class-based (vs legacy) element.
+  let fullName;      // Fully qualified reference, e.g. `Polymer.DomModule`.
+  let className;     // Just the class name e.g. `DomModule`.
+  let parent;        // Where in the namespace tree does this live.
 
   if (feature.className) {
+    isClassBased = true;
```

Comment:
Question for @rictic. It seems that all elements have the `class` kind, no matter if they are classes or not. So the only way I've found to tell the difference is that real classes have `className`. Does that sound like the right way to do this?

Reply:
I think the factor that determines whether to emit an interface or class should be whether the element's constructor is assigned into a namespace, not whether it uses legacy syntax or now.

Reply:
Also added some TODOs around behaviors/mixins. How do we handle behaviors when we are emitting classes? And how do we handle mixins when we are emitting an interface? We don't currently define interfaces for mixins (they are functions), so we can't just add them to extends. Maybe we need to emit interfaces for mixins too.